test/behaviour/graql/GraqlSteps.java

```diff
@@ -22,39 +22,47 @@
 import grakn.client.concept.Concept;
 import grakn.client.concept.answer.ConceptMap;
 import grakn.client.concept.answer.ConceptMapGroup;
+import grakn.client.concept.answer.Numeric;
 import grakn.client.concept.answer.NumericGroup;
 import grakn.client.concept.thing.Attribute;
+import grakn.client.concept.type.AttributeType;
 import grakn.client.rpc.QueryFuture;
 import graql.lang.Graql;
 import graql.lang.query.GraqlDefine;
 import graql.lang.query.GraqlDelete;
 import graql.lang.query.GraqlInsert;
+import graql.lang.query.GraqlMatch;
 import graql.lang.query.GraqlUndefine;
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
 
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import static grakn.client.test.behaviour.connection.ConnectionSteps.tx;
 import static grakn.client.test.behaviour.util.Util.assertThrows;
 import static grakn.client.test.behaviour.util.Util.assertThrowsWithMessage;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
 
 public class GraqlSteps {
 
-    private static Stream<ConceptMap> answers;
-    private static Number numericAnswer;
+    private static List<ConceptMap> answers;
```

Comment:
`Stream<ConceptMap>` was used for efficiency: allowing us to asynchronously run `match` while executing other steps. However, if we wanted to execute two follow-up steps (e.g: identify the answers in two different ways), that would require re-executing the query.

To mitigate this issue, we now collect the entire `Stream` to a `List` in `get answers of graql match`.